Project1/Map.cpp

```diff
@@ -16,30 +16,31 @@ void Map::insertAdjacencies() {
 	for (int x = 0; x < width; x++) {
 		for (int y = 0; y < height; y++) {
 			if (x > 0) {
-				_grid[x][y].addAdjacentCell(&(_grid[x - 1][y]));
+				_grid[x][y].addAdjacentCell(_grid[x - 1][y]);
 			} if (x < width - 1) {
-				_grid[x][y].addAdjacentCell(&(_grid[x + 1][y]));
+				_grid[x][y].addAdjacentCell(_grid[x + 1][y]);
 			}if (y > 0) {
-				_grid[x][y].addAdjacentCell(&(_grid[x][y - 1]));
+				_grid[x][y].addAdjacentCell(_grid[x][y - 1]);
 			} if (y < height - 1) {
-				_grid[x][y].addAdjacentCell(&(_grid[x][y + 1]));
+				_grid[x][y].addAdjacentCell(_grid[x][y + 1]);
 			}
 		}
 	}
 }
-void Map::moveUnit(GridCell* start, GridCell* destination) {
-	Expects(start->getTile().hasUnit() && !destination->getTile().hasUnit());
-	_unit_to_cell.at(start->getTile()._unit) = destination; // Map the Unit to the destination.
-	destination->getTile().insertUnit(start->getTile().removeUnit()); // Put the Unit in the destination Tile
+void Map::moveUnit(GridCell& start, GridCell& destination) {
+	Expects(start.getTile().hasUnit() && !destination.getTile().hasUnit());
+	Unit* unit = start.getTile()._unit;
+	removeUnit(unit);
+	insertUnit(unit, destination);
 }
-void Map::insertUnit(Unit* new_unit, GridCell* destination) {
-	Expects(!destination->getTile().hasUnit());
-	_unit_to_cell.insert(std::pair<Unit*, GridCell*>(new_unit, destination));// Map the Unit to the destination.
-	destination->getTile().insertUnit(new_unit);
+void Map::insertUnit(Unit* new_unit, GridCell& destination) {
+	Expects(!destination.getTile().hasUnit());
```

Comment:
I'm assuming you don't want anyone to be inserting nullptr?

You should make it take a Unit& or add != nullptr to the Expects